In [ ]:
# %pip install pandas openpyxl 
#   use of excel writer

In [1]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

import pingouin as pg

In [ ]:
#   File Beer Profiles and ratings.csv
#   --------------------------------------

# This file contains 1 Tabs  with Beer dateset detailled profile and ratings info
    #   Objective of this script - check relations between dimension concepts


In [2]:
# Fetch raw data

rawdata = pd.read_csv(r'C:\Users\t.vincotte\Documents\CODE\TFE\EoF - Work\Beer_Profile\Source_Files\beer_data_set.csv') 
 

In [ ]:
rawdata.count()

In [4]:
#rawdata.info()
str = rawdata.info()
structure = pd.DataFrame(str)  #  ne met pas en dataframe
structure
# structure.to_excel("Beer_datasets_structures",sheet_name='Beer data set details') # n ecrit pas le rawdata.info dans le fichier

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5558 entries, 0 to 5557
Data columns (total 21 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         5556 non-null   object 
 1   key          5558 non-null   int64  
 2   Style        5558 non-null   object 
 3   Style Key    5558 non-null   int64  
 4   Brewery      5558 non-null   object 
 5   Description  5558 non-null   object 
 6   ABV          5558 non-null   float64
 7   Ave Rating   5558 non-null   float64
 8   Min IBU      5558 non-null   int64  
 9   Max IBU      5558 non-null   int64  
 10  Astringency  5558 non-null   int64  
 11  Body         5558 non-null   int64  
 12  Alcohol      5558 non-null   int64  
 13  Bitter       5558 non-null   int64  
 14  Sweet        5558 non-null   int64  
 15  Sour         5558 non-null   int64  
 16  Salty        5558 non-null   int64  
 17  Fruits       5558 non-null   int64  
 18  Hoppy        5558 non-null   int64  
 19  Spices

Empty DataFrame
Columns: []
Index: []

In [5]:
rawdata.head(1)

Name  key    Style  Style Key              Brewery  \
0  Amber  251  Altbier          8  Alaskan Brewing Co.   

                                         Description  ABV  Ave Rating  \
0  Notes:Richly malty and long on the palate, wit...  5.3        3.65   

   Min IBU  Max IBU  ...  Body  Alcohol  Bitter  Sweet  Sour  Salty  Fruits  \
0       25       50  ...    32        9      47     74    33      0      33   

   Hoppy  Spices  Malty  
0     57       8    111  

[1 rows x 21 columns]

In [ ]:
#   Only show columns we want to see - methods not used here
#   ----------------------------------------------------------

        # Using DataFrame.drop
            #   df.drop(df.columns[[1, 2]], axis=1, inplace=True)

        # Select the ones you want
            #   df1 = df[['a','d']]


In [17]:
#   Beer dimension check - rough listing - create spreadsheet 
#   ----------------------------------------------------------

    #   https://www.geeksforgeeks.org/exporting-a-pandas-dataframe-to-an-excel-file/
    #   https://stackoverflow.com/questions/42370977/how-to-save-a-new-sheet-in-an-existing-excel-file-using-pandas
    #   https://saturncloud.io/blog/how-to-add-a-worksheet-to-an-existing-excel-file-with-pandas/ 

df1 = rawdata.drop(['Style Key', 'Description'], axis=1)
df1.to_excel("EDA_DimCheck_BeerDataset.xlsx",sheet_name='rough_listing')

        #   Note : (Re)creates spreadsheet as running - below - use method to append sheet


Alternate method (stack overflow) - Add sheet to existing file
-------------------------------------
import pandas as pd
import numpy as np

path = r"C:\Users\fedel\Desktop\excelData\PhD_data.xlsx"

x3 = np.random.randn(100, 2)
df3 = pd.DataFrame(x3)

x4 = np.random.randn(100, 2)
df4 = pd.DataFrame(x4)

writer = pd.ExcelWriter(path, engine = 'xlsxwriter')
df3.to_excel(writer, sheet_name = 'x3')
df4.to_excel(writer, sheet_name = 'x4')
writer.save()
writer.close()

----

import pandas as pd

data = {'EmployeeID': [101, 102, 103],
        'EmployeeName': ['Alice', 'Bob', 'Charlie'],
        'Salary': [60000, 70000, 80000]}

df_new = pd.DataFrame(data)

with pd.ExcelWriter('existing_file.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df_new.to_excel(writer, sheet_name='New Sheet', index=False)

In [18]:
#   Beer dimension check - List by style
#   ---------------------------------------
    #   Use method to append sheet to existing excel file

df1_by_style = df1[['Style','Name','Brewery','ABV']].sort_values(by=['Name']).sort_values(by=['Style'])
with pd.ExcelWriter('EDA_DimCheck_BeerDataset.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df1_by_style.to_excel(writer, sheet_name='list_by_style', index=True)

# df1_by_style.to_excel("EDA_DimCheck_BeerDataset.xlsx",sheet_name='list_by_style')


In [ ]:
#   Beer dimension check - List by brewery
#   ---------------------------------------
   #   Use method to append sheet to existing excel file

df1_by_brewery = df1[['Brewery','Name','Style','ABV']].sort_values(by=['Name']).sort_values(by=['Brewery'])
with pd.ExcelWriter('EDA_DimCheck_BeerDataset.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df1_by_brewery.to_excel(writer, sheet_name='list_by_brewery', index=True)


In [23]:
#   Beer dimension check - List by style_brewery
#   ---------------------------------------------
   #   Use method to append sheet to existing excel file

df1_by_brewery_style = df1[['Style','Brewery','Name','ABV']].sort_values(by=['Name']).sort_values(by=['Brewery']).sort_values(by=['Style'])
with pd.ExcelWriter('EDA_DimCheck_BeerDataset.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df1_by_brewery_style.to_excel(writer, sheet_name='list_by_brewery_style', index=True)


In [24]:
#   Beer dimension check - List by brewery_style
#   ---------------------------------------------
   #   Use method to append sheet to existing excel file

df1_by_brewery_style = df1[['Brewery','Style','Name','ABV']].sort_values(by=['Name']).sort_values(by=['Style']).sort_values(by=['Brewery'])
with pd.ExcelWriter('EDA_DimCheck_BeerDataset.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df1_by_brewery_style.to_excel(writer, sheet_name='list_by_style_brewery', index=True)


In [35]:
#     Check if dupplicate beer names
#   -------------------------------------
#   
df1_duplicate_beer_check = df1[['Name','ABV']].count()
df1_duplicate_beer_check 


Name    5556
ABV     5558
dtype: int64

In [44]:
#     Check if dupplicate beer names
#   -------------------------------------
#   
 
df2x = df1[['Name']].drop_duplicates
#df2 = pd.DataFrame[df2x]
df2x

#df2_duplicate_beer_check = df2.count()
#df2_duplicate_beer_check 

<bound method DataFrame.drop_duplicates of                                              Name
0                                           Amber
1                                      Double Bag
2                                  Long Trail Ale
3                                    Doppelsticke
4                                          Scurry
...                                           ...
5553                          Red Nose Winter Ale
5554  Our Special Ale 2019 (Anchor Christmas Ale)
5555                         Fish Tale Winterfish
5556    He'Brew Hanukkah, Chanukah: Pass The Beer
5557     Frosted Frog Christmas Ale - Barrel-Aged

[5558 rows x 1 columns]>

Afre drop dupplicate still amount of rows >> no duplicate in beer name

Index count -5556 - see 2 rows where name starts with # while we have others containing # in the name
    row 82 and row 1250
Checked - (see here below) - name are well # 100 and #9 - no lost info due to special char #

??? Any specialchar affecting the count value ?? 
??  Need to clean name for further treatment ??

In [29]:
df1.loc[[82]]
df1.loc[[1250]]

Name   key                 Style                    Brewery  ABV  \
1250   #9  3301  Fruit and Field Beer  Magic Hat Brewing Company  5.1   

      Ave Rating  Min IBU  Max IBU  Astringency  Body  Alcohol  Bitter  Sweet  \
1250        3.42        5       45           22    23        1      31     73   

      Sour  Salty  Fruits  Hoppy  Spices  Malty  
1250    38      0      74     50       2     28

5558 records